src/services/state-persistor/state-persistor-featureinfo.service.ts

```diff
@@ -0,0 +1,28 @@
+import { SP_KEY_FID } from './state-persistor.model'
+import { storageHelper } from './storage/storage.helper'
+import { useFeatureInfoStore } from '@/stores/feature-info.store'
+
+class StatePersistorFeatureInfoService {
+  bootstrap() {
+    this.restore()
+  }
+
+  restore() {
+    const fid = <string | undefined>storageHelper.getValue(SP_KEY_FID)
+
+    if (fid) {
+      const { setFid } = useFeatureInfoStore()
+      setFid(fid)
+      this.removeQueryParam(SP_KEY_FID)
+    }
+  }
+
+  removeQueryParam(param: string): void {
```

Comment:
maybe this one is a utility and should be in `urlStorage.ts` helper?

Reply:
Good point, I've moved it.